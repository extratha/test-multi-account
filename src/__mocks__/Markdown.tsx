const MockMarkdown = (props: any) => {
  return <div {...props} />;
};

jest.mock("react-markdown", () => MockMarkdown);
