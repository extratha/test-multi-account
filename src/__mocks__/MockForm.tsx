import { yupResolver } from "@hookform/resolvers/yup";
import { ReactNode, useEffect } from "react";
import { FormProvider, UseFormReturn, useForm } from "react-hook-form";
import { ObjectSchema } from "yup";

export interface MockFormProps<T = any> {
  initValues: T | undefined;
  resolver?: ObjectSchema<any>;
  children?: ReactNode;
  onInitForm?: (methods: UseFormReturn) => void;
  onSubmit: () => void;
}

export interface MockHandleValueProps {
  name: string;
}

const MockForm = (props: MockFormProps) => {
  const { initValues, resolver, children, onInitForm, onSubmit } = props;
  const methods = useForm({
    reValidateMode: "onSubmit",
    defaultValues: initValues,
    resolver: resolver ? yupResolver(resolver) : undefined,
  });

  useEffect(() => {
    if (onInitForm) onInitForm(methods);
  }, []);

  return (
    <FormProvider {...methods}>
      <form noValidate onSubmit={methods.handleSubmit(onSubmit)}>
        {children}
        <button type="submit" data-testid="submit-button">
          Submit
        </button>
      </form>
    </FormProvider>
  );
};

export default MockForm;
