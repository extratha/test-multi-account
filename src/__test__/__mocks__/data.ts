export const data = {
  test: 'test'
}