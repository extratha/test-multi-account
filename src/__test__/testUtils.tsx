import { RenderOptions, RenderResult, render } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { SWRConfig } from 'swr';
import { theme } from '@/config/config-mui/theme';
import { ThemeProvider as MUIThemeProvider } from '@mui/material/styles';
import { NextIntlClientProvider } from 'next-intl';
import commonTh from '../../public/locales/th/common.json';
import commonEn from '../../public/locales/th/common.json';
type RenderWithProvider = (
  elm: React.ReactElement,
  renderOptions?: ProviderOptions,
) => RenderResult;

export type ProviderOptions = {
  formOptions?: Record<string, unknown>;
} & RenderOptions;

const messages = {
  ...commonTh,
  ...commonEn,
};

const renderWithProviders: RenderWithProvider = (
  component,
  renderOptions?,
  locale = 'th',
) => {
  const wrapper = ({ children }: { children: React.ReactNode }) => (
    <NextIntlClientProvider locale={locale} messages={messages}>
      <MUIThemeProvider theme={theme}>
        <SWRConfig value={{ provider: () => new Map() }}>{children}</SWRConfig>
      </MUIThemeProvider>
    </NextIntlClientProvider>
  );

  return render(component, {
    wrapper,
    ...renderOptions,
  });
};

export { renderWithProviders as render, userEvent };
export * from '@testing-library/react';
