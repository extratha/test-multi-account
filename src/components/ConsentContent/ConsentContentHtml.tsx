import { styled } from "@mui/material";

export interface ConsentContentHtmlProps {
  name: string;
  html: string;
}

const Html = styled("div")({
  fontSize: "16px",
  lineHeight: "28px",
  "& > h3, & > p": {
    "&:first-of-type": {
      marginTop: "0px",
    },
    "&:last-of-type": {
      marginBottom: "0px",
    },
  },
  "& ul": {
    paddingLeft: "24px",
    listStyleType: "disc",
  },
  "& li": {
    margin: "16px 0px",
  },
  "& ol": {
    listStyleType: "none",
    counterReset: "item",
    padding: "0px",
    "& > li": {
      display: "table-row",
      counterIncrement: "item",
      borderCollapse: "separate",
      borderSpacing: "0px 6px",
      "&:before": {
        display: "table-cell",
        content: `counters(item, ".") ". "`,
        paddingRight: "4px",
        whiteSpace: "nowrap",
      },
    },
    "& > ol, & > ul": {
      paddingLeft: "24px",
      "& > li > ol, & > li > ul": {
        padding: "0px",
      },
    },
  },
  "& table": {
    width: "100%",
    borderCollapse: "collapse",
    tableLayout: "fixed",
    margin: "16px 0px",
    "& li": {
      margin: "0px",
    },
  },
  "& th": {
    height: "48px",
    textAlign: "center",
    color: "#ffffff",
    border: "1px solid #000000",
    borderRightColor: "#ffffff",
    backgroundColor: "#002060",
    "&:last-child": {
      borderColor: "#000000",
    },
  },
  "& td": {
    border: "1px solid #000000",
    padding: "8px",
    verticalAlign: "top",
  },
});

const ConsentContentHtml = ({ name, html }: ConsentContentHtmlProps) => {
  return (
    <Html
      data-testid={`${name}-content`}
      dangerouslySetInnerHTML={{
        __html: html,
      }}
    />
  );
};

export default ConsentContentHtml;