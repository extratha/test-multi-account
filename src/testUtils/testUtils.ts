import { act, render as testRender } from "@testing-library/react";
import dayjs from "dayjs";
import flushPromises from "flush-promises";
import { ReactElement } from "react";

import { TestProvider } from "./TestProvider";

export const API = {
  LOGIN: "/auth/login",
  EXAMPLE_AI_INTERPRET_URL: "/lab/examples/labId",
  GET_LAB_INTERPRETS_ID_URL: "/lab/interprets/labId",
  SUBMIT_LAB_INTERPRETS_URL: "/lab/interprets",
  CONSENT: "/consents",
  TERMS_AND_CONDITIONS: "/consents/terms-conditions",
  TERMS_AND_CONDITIONS_LATEST: "/consents/terms-conditions/latest",
  PRIVACY_POLICY: "/consents/privacy-policies",
  PRIVACY_POLICY_LATEST: "/consents/privacy-policies/latest",
  PUBLIC_TERMS_CONDITIONS: "/consents/terms-conditions/public",
  PUBLIC_PRIVACY_POLICIES: "/consents/privacy-policies/public",
};

export const render = (ui: ReactElement) => {
  return testRender(ui, { wrapper: TestProvider });
};

export const flushPromise = async () => {
  await act(async () => {
    await flushPromises();
  });
};

export const advanceTimersByTime = async (time: number) => {
  await act(async () => {
    jest.advanceTimersByTime(time);
  });
};

export const setMockDate = (date: string) => {
  jest.setSystemTime(dayjs(date).valueOf());
};
