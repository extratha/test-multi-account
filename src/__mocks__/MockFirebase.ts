import { mockInputFieldConfig } from "./data";

jest.mock("firebase/app", () => ({
  initializeApp: jest.fn(),
}));

jest.mock("firebase/remote-config", () => ({
  getRemoteConfig: jest.fn().mockReturnValue({ settings: {} }),
  fetchAndActivate: jest.fn().mockResolvedValue(""),
  getValue: jest.fn().mockReturnValue({
    asString: () => mockInputFieldConfig,
  }),
}));
