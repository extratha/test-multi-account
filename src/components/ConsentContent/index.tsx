import { Box, Collapse, IconButton, Stack, Typography, styled } from "@mui/material";
import { useState } from "react";

import { IconChevronDown, IconChevronUp, IconConsentGroupDetail } from "@/assets";
import useTranslation from "@/locales/useLocale";
import { ConsentData } from "@/types/model.api";
import ConsentContentHtml from "./ConsentContentHtml";

export interface ConsentContentProps {
  name: string;
  data: ConsentData;
}

const ContentSection = styled(Stack)(({ theme }) => ({
  padding: "24px",
  borderRadius: "12px",
  backgroundColor: theme.palette.surfaceGray.lowest,
}));

const ServiceItem = styled(Stack)(({ theme }) => ({
  border: `1px solid ${theme.palette.background.border}`,
  backgroundColor: theme.palette.background.paper,
  "&:first-of-type": {
    borderRadius: "12px 12px 0 0",
  },
  "&:last-of-type": {
    borderRadius: "0 0 12px 12px",
  },
}));

const ServiceHeader = styled(Stack)({
  cursor: "pointer",
  flexDirection: "row",
  padding: "24px",
});

const ServiceContent = styled(Stack)({
  padding: "16px",
  // borderTop: `1px solid ${NEUTRAL[80]}`,
});

const ServiceLogo = styled("img")({
  maxWidth: "52px",
  maxHeight: "52px",
});

const ArrowUp = styled(IconChevronUp)({
  width: "24px",
  height: "24px",
  // color: CUSTOM_COLORS.lightSteelgray,
});

const ArrowDown = styled(IconChevronDown)({
  width: "24px",
  height: "24px",
  // color: CUSTOM_COLORS.lightSteelgray,
});

const ConsentContent = (props: ConsentContentProps) => {
  const { name, data } = props;
  const { translation } = useTranslation();

  const [isExpandServices, setIsExpandServices] = useState<boolean[]>([]);

  const handleClickService = (serviceIndex: number) => {
    setIsExpandServices((prevState) => {
      const newState = [...prevState];
      newState[serviceIndex] = !newState[serviceIndex];
      return newState;
    });
  };

  return (
    <Stack spacing={3}>
      <ContentSection>
        <ConsentContentHtml name={name} html={data.content} />
      </ContentSection>
      {data.services.length > 0 && (
        <ContentSection data-testid={`${name}-service`}>
          <Typography variant="titleBold" marginBottom="24px">
            {translation("Common.title.allServices")}
          </Typography>
          {data.services.map((service, index) => (
            <ServiceItem key={index}>
              <ServiceHeader data-testid={`${name}-service-${index}`} onClick={() => handleClickService(index)}>
                <Box>
                  <ServiceLogo src={service.logo} data-testid={`${name}-service-${index}-logo`} alt="" />
                </Box>
                <Stack flex="1" spacing="4px" marginLeft="12px">
                  <Typography variant="bodyMedium" data-testid={`${name}-service-${index}-title`}>
                    {service.title}
                  </Typography>
                  <IconConsentGroupDetail data-testid={`${name}-service-${index}-detail-icon`} />
                </Stack>
                <Stack justifyContent="center">
                  <IconButton>{isExpandServices[index] === true ? <ArrowUp /> : <ArrowDown />}</IconButton>
                </Stack>
              </ServiceHeader>
              <Collapse in={isExpandServices[index]}>
                <ServiceContent data-testid={`${name}-service-${index}-content`}>
                  <ConsentContentHtml name={name} html={service.content} />
                </ServiceContent>
              </Collapse>
            </ServiceItem>
          ))}
        </ContentSection>
      )}
    </Stack>
  );
};

export default ConsentContent;
