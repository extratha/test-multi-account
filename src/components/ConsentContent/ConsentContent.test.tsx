import clone from "clone";

import { mockTermsAndConsData } from "@/__mocks__/data";
import { render, screen, userEvent, waitFor } from "@/testUtils/testUtils";
import ConsentContent, { ConsentContentProps } from ".";

describe("ConsentContent", () => {
  let props: ConsentContentProps;

  beforeEach(() => {
    props = { name: "test", data: clone(mockTermsAndConsData.consent) };
  });

  const renderConsentContent = () => {
    return render(<ConsentContent {...props} />);
  };

  it("should render correctly", () => {
    const { asFragment } = renderConsentContent();
    expect(asFragment()).toMatchSnapshot();
  });

  it("should not display services consent when services are empty", () => {
    props.data.services = [];
    renderConsentContent();

    expect(screen.queryByTestId("test-service")).not.toBeInTheDocument();
  });

  it("should collapse/expand service content when click service header", async () => {
    renderConsentContent();

    expect(screen.getByTestId("test-service-0-content")).not.toBeVisible();
    expect(screen.getByTestId("test-service-1-content")).not.toBeVisible();
    expect(screen.getByTestId("test-service-2-content")).not.toBeVisible();
    expect(screen.getByTestId("test-service-3-content")).not.toBeVisible();

    await userEvent.click(screen.getByTestId("test-service-0"));

    expect(screen.getByTestId("test-service-0-content")).toBeVisible();
    expect(screen.getByTestId("test-service-1-content")).not.toBeVisible();
    expect(screen.getByTestId("test-service-2-content")).not.toBeVisible();
    expect(screen.getByTestId("test-service-3-content")).not.toBeVisible();

    await userEvent.click(screen.getByTestId("test-service-1"));

    expect(screen.getByTestId("test-service-0-content")).toBeVisible();
    expect(screen.getByTestId("test-service-1-content")).toBeVisible();
    expect(screen.getByTestId("test-service-2-content")).not.toBeVisible();
    expect(screen.getByTestId("test-service-3-content")).not.toBeVisible();

    await userEvent.click(screen.getByTestId("test-service-0"));

    await waitFor(() => expect(screen.getByTestId("test-service-0-content")).not.toBeVisible());
    expect(screen.getByTestId("test-service-1-content")).toBeVisible();
    expect(screen.getByTestId("test-service-2-content")).not.toBeVisible();
    expect(screen.getByTestId("test-service-3-content")).not.toBeVisible();
  });
});
