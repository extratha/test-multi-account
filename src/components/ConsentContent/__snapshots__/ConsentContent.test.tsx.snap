// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConsentContent should render correctly 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.emotion-0>:not(style):not(style) {
  margin: 0;
}

.emotion-0>:not(style)~:not(style) {
  margin-top: 24px;
}

.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 24px;
  border-radius: 12px;
  background-color: #FAFAFA;
}

.emotion-2 {
  font-size: 16px;
  line-height: 28px;
}

.emotion-2>h3:first-of-type,
.emotion-2>p:first-of-type {
  margin-top: 0px;
}

.emotion-2>h3:last-of-type,
.emotion-2>p:last-of-type {
  margin-bottom: 0px;
}

.emotion-2 ul {
  padding-left: 24px;
  list-style-type: disc;
}

.emotion-2 ol {
  list-style-type: none;
  counter-reset: item;
  padding: 0px;
}

.emotion-2 ol>li {
  display: table-row;
  counter-increment: item;
  border-collapse: separate;
  border-spacing: 0px 6px;
}

.emotion-2 ol>li:before {
  display: table-cell;
  content: counters(item, ".") ". ";
  padding-right: 6px;
  white-space: nowrap;
}

.emotion-2 ol>ol,
.emotion-2 ol>ul {
  padding-left: 24px;
}

.emotion-2 ol>ol>li>ol,
.emotion-2 ol>ul>li>ol,
.emotion-2 ol>ol>li>ul,
.emotion-2 ol>ul>li>ul {
  padding: 0px;
}

.emotion-2 li {
  margin: 16px 0px;
}

.emotion-2 li ol>li {
  border-collapse: separate;
  border-spacing: 0px 6px;
}

.emotion-2 li ol>li:before {
  content: counters(item, ".") " ";
  padding-right: 6px;
}

.emotion-2 table {
  width: 100%;
  border-collapse: collapse;
  table-layout: fixed;
  margin: 16px 0px;
}

.emotion-2 table li {
  margin: 0px;
}

.emotion-2 th {
  height: 48px;
  text-align: center;
  color: #ffffff;
  border: 1px solid #000000;
  border-right-color: #ffffff;
  background-color: #002060;
}

.emotion-2 th:last-child {
  border-color: #000000;
}

.emotion-2 td {
  border: 1px solid #000000;
  padding: 8px;
  vertical-align: top;
}

.emotion-4 {
  margin: 0;
  font-size: 18px;
  font-weight: 700;
  line-height: 24px;
  margin-bottom: 24px;
}

.emotion-5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  border: 1px solid #cfd8dc;
  background-color: #fff;
}

.emotion-5:first-of-type {
  border-radius: 12px 12px 0 0;
}

.emotion-5:last-of-type {
  border-radius: 0 0 12px 12px;
}

.emotion-6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  cursor: pointer;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  padding: 24px;
}

.emotion-8 {
  max-width: 52px;
  max-height: 52px;
}

.emotion-9 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  margin-left: 12px;
}

.emotion-9>:not(style):not(style) {
  margin: 0;
}

.emotion-9>:not(style)~:not(style) {
  margin-top: 4px;
}

.emotion-10 {
  margin: 0;
  font-size: 16px;
  font-weight: 500;
  line-height: 1.5;
}

.emotion-11 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
}

.emotion-12 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  position: relative;
  box-sizing: border-box;
  -webkit-tap-highlight-color: transparent;
  background-color: transparent;
  outline: 0;
  border: 0;
  margin: 0;
  border-radius: 0;
  padding: 0;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
  -moz-appearance: none;
  -webkit-appearance: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
  text-align: center;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  font-size: 1.5rem;
  padding: 8px;
  border-radius: 50%;
  overflow: visible;
  color: rgba(0, 0, 0, 0.54);
  -webkit-transition: background-color 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: background-color 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
}

.emotion-12::-moz-focus-inner {
  border-style: none;
}

.emotion-12.Mui-disabled {
  pointer-events: none;
  cursor: default;
}

@media print {
  .emotion-12 {
    -webkit-print-color-adjust: exact;
    color-adjust: exact;
  }
}

.emotion-12:hover {
  background-color: rgba(0, 0, 0, 0.04);
}

@media (hover: none) {
  .emotion-12:hover {
    background-color: transparent;
  }
}

.emotion-12.Mui-disabled {
  background-color: transparent;
  color: rgba(0, 0, 0, 0.26);
}

.emotion-13 {
  width: 24px;
  height: 24px;
}

.emotion-14 {
  overflow: hidden;
  pointer-events: none;
  position: absolute;
  z-index: 0;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  border-radius: inherit;
}

.emotion-15 {
  height: 0;
  overflow: hidden;
  -webkit-transition: height 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: height 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  visibility: hidden;
}

.emotion-16 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
}

.emotion-17 {
  width: 100%;
}

.emotion-18 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 16px;
}

<div
    class="MuiStack-root emotion-0"
  >
    <div
      class="MuiStack-root emotion-1"
    >
      <div
        class="emotion-2"
        data-testid="test-content"
      >
        **Terms of Service**

Welcome to our service. By using our service, you agree to the following terms...

**Privacy Policy**

We value your privacy. This policy explains how we collect and use your information...
      </div>
    </div>
    <div
      class="MuiStack-root emotion-1"
      data-testid="test-service"
    >
      <span
        class="MuiTypography-root MuiTypography-titleBold emotion-4"
      >
        All Services
      </span>
      <div
        class="MuiStack-root emotion-5"
      >
        <div
          class="MuiStack-root emotion-6"
          data-testid="test-service-0"
        >
          <div
            class="MuiBox-root emotion-7"
          >
            <img
              alt=""
              class="emotion-8"
              data-testid="test-service-0-logo"
              src="https://cdn-dev-playground.cariva.co.th/terms/images/logo.png"
            />
          </div>
          <div
            class="MuiStack-root emotion-9"
          >
            <span
              class="MuiTypography-root MuiTypography-bodyMedium emotion-10"
              data-testid="test-service-0-title"
            >
              mock-test-title
            </span>
            <svg
              data-testid="test-service-0-detail-icon"
            />
          </div>
          <div
            class="MuiStack-root emotion-11"
          >
            <button
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium emotion-12"
              tabindex="0"
              type="button"
            >
              <svg
                class="emotion-13"
              />
              <span
                class="MuiTouchRipple-root emotion-14"
              />
            </button>
          </div>
        </div>
        <div
          class="MuiCollapse-root MuiCollapse-vertical MuiCollapse-hidden emotion-15"
          style="min-height: 0px;"
        >
          <div
            class="MuiCollapse-wrapper MuiCollapse-vertical emotion-16"
          >
            <div
              class="MuiCollapse-wrapperInner MuiCollapse-vertical emotion-17"
            >
              <div
                class="MuiStack-root emotion-18"
                data-testid="test-service-0-content"
              >
                <div
                  class="emotion-2"
                  data-testid="test-content"
                >
                  **Service Overview**

AI Interpret provides advanced language translation capabilities...

**Features**

- Real-time translation
- Multi-language support
- High accuracy rates

**Usage Instructions**

1. Select your input and output languages.
2. Speak or type your text.
3. Receive the translated output.
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="MuiStack-root emotion-5"
      >
        <div
          class="MuiStack-root emotion-6"
          data-testid="test-service-1"
        >
          <div
            class="MuiBox-root emotion-7"
          >
            <img
              alt=""
              class="emotion-8"
              data-testid="test-service-1-logo"
              src="https://cdn-dev-playground.cariva.co.th/terms/images/logo.png"
            />
          </div>
          <div
            class="MuiStack-root emotion-9"
          >
            <span
              class="MuiTypography-root MuiTypography-bodyMedium emotion-10"
              data-testid="test-service-1-title"
            >
              Health Visualization
            </span>
            <svg
              data-testid="test-service-1-detail-icon"
            />
          </div>
          <div
            class="MuiStack-root emotion-11"
          >
            <button
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium emotion-12"
              tabindex="0"
              type="button"
            >
              <svg
                class="emotion-13"
              />
              <span
                class="MuiTouchRipple-root emotion-14"
              />
            </button>
          </div>
        </div>
        <div
          class="MuiCollapse-root MuiCollapse-vertical MuiCollapse-hidden emotion-15"
          style="min-height: 0px;"
        >
          <div
            class="MuiCollapse-wrapper MuiCollapse-vertical emotion-16"
          >
            <div
              class="MuiCollapse-wrapperInner MuiCollapse-vertical emotion-17"
            >
              <div
                class="MuiStack-root emotion-18"
                data-testid="test-service-1-content"
              >
                <div
                  class="emotion-2"
                  data-testid="test-content"
                >
                  **Service Overview**

Health Visualization helps you understand your health data through visual representations...

**Features**

- Detailed health metrics
- Trend analysis
- Personalized insights

**Usage Instructions**

1. Upload your health data.
2. Choose the metrics you want to visualize.
3. View the visualized data and insights.
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="MuiStack-root emotion-5"
      >
        <div
          class="MuiStack-root emotion-6"
          data-testid="test-service-2"
        >
          <div
            class="MuiBox-root emotion-7"
          >
            <img
              alt=""
              class="emotion-8"
              data-testid="test-service-2-logo"
              src="https://cdn-dev-playground.cariva.co.th/terms/images/logo.png"
            />
          </div>
          <div
            class="MuiStack-root emotion-9"
          >
            <span
              class="MuiTypography-root MuiTypography-bodyMedium emotion-10"
              data-testid="test-service-2-title"
            >
              ASR
            </span>
            <svg
              data-testid="test-service-2-detail-icon"
            />
          </div>
          <div
            class="MuiStack-root emotion-11"
          >
            <button
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium emotion-12"
              tabindex="0"
              type="button"
            >
              <svg
                class="emotion-13"
              />
              <span
                class="MuiTouchRipple-root emotion-14"
              />
            </button>
          </div>
        </div>
        <div
          class="MuiCollapse-root MuiCollapse-vertical MuiCollapse-hidden emotion-15"
          style="min-height: 0px;"
        >
          <div
            class="MuiCollapse-wrapper MuiCollapse-vertical emotion-16"
          >
            <div
              class="MuiCollapse-wrapperInner MuiCollapse-vertical emotion-17"
            >
              <div
                class="MuiStack-root emotion-18"
                data-testid="test-service-2-content"
              >
                <div
                  class="emotion-2"
                  data-testid="test-content"
                >
                  **Service Overview**

ASR (Automatic Speech Recognition) converts spoken language into text...

**Features**

- High accuracy speech-to-text
- Supports multiple languages
- Real-time processing

**Usage Instructions**

1. Select your language.
2. Speak into the microphone.
3. View the transcribed text.
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="MuiStack-root emotion-5"
      >
        <div
          class="MuiStack-root emotion-6"
          data-testid="test-service-3"
        >
          <div
            class="MuiBox-root emotion-7"
          >
            <img
              alt=""
              class="emotion-8"
              data-testid="test-service-3-logo"
              src="https://cdn-dev-playground.cariva.co.th/terms/images/logo.png"
            />
          </div>
          <div
            class="MuiStack-root emotion-9"
          >
            <span
              class="MuiTypography-root MuiTypography-bodyMedium emotion-10"
              data-testid="test-service-3-title"
            >
              Symptom Checker
            </span>
            <svg
              data-testid="test-service-3-detail-icon"
            />
          </div>
          <div
            class="MuiStack-root emotion-11"
          >
            <button
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium emotion-12"
              tabindex="0"
              type="button"
            >
              <svg
                class="emotion-13"
              />
              <span
                class="MuiTouchRipple-root emotion-14"
              />
            </button>
          </div>
        </div>
        <div
          class="MuiCollapse-root MuiCollapse-vertical MuiCollapse-hidden emotion-15"
          style="min-height: 0px;"
        >
          <div
            class="MuiCollapse-wrapper MuiCollapse-vertical emotion-16"
          >
            <div
              class="MuiCollapse-wrapperInner MuiCollapse-vertical emotion-17"
            >
              <div
                class="MuiStack-root emotion-18"
                data-testid="test-service-3-content"
              >
                <div
                  class="emotion-2"
                  data-testid="test-content"
                >
                  **Service Overview**

Symptom Checker helps you understand potential causes for your symptoms...

**Features**

- Symptom input
- Potential diagnoses
- Suggested next steps

**Usage Instructions**

1. Enter your symptoms.
2. View the potential diagnoses.
3. Follow the suggested next steps for further action.
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
